### PR TITLE
mgr: add new role for ceph-mgr

### PR DESCRIPTION
**Pull request by leseb**
The Ceph Manager daemon (ceph-mgr) runs alongside monitor daemons, to
provide additional monitoring and interfaces to external monitoring and
management systems.

Only works as of the Kraken release.

Co-Authored-By: Guillaume Abrioux <gabrioux@redhat.com>
Signed-off-by: Sébastien Han <seb@redhat.com>

### CODE REVIEW

**Comment by leseb** [comment]
jenkins test ansible2.2-xenial_cluster

**Comment by leseb** [comment]
jenkins test Testing: ansible2.2-docker_dmcrypt_journal_collocation

**Comment by leseb** [comment]
jenkins test ansible2.2-docker_dedicated_journal

**Comment by leseb** [comment]
jenkins test ansible2.2-docker_cluster

**Comment by leseb** [comment]
jenkins test ansible2.2-docker_dmcrypt_journal_collocation

**Comment by leseb** [comment]
We need to enable Kraken in Ceph Ansible for this to work.

**Comment by leseb** [comment]
@batrick thanks for the suggestion. I think I've seen that too on some containerized scenario too. Let's remove the `run_once` and run this on a single monitor.

**Comment by batrick** [comment]
@leseb, been playing around with this some more on Linode. I found that some mgr modules depend on a few packages that aren't installed. Here's what I had to install manually:

fsstatus module => python-prettytable
rest module => python-cherrpy python-django python-dateutil

I dunno if you'll want to add those in or not since they are just modules and not essential to the base mgr.

File: roles/ceph-mon/tasks/docker/main.yml [hunk 1]
```diff
@@ -88,3 +88,17 @@
...
+    creates: /etc/ceph/{{ cluster }}.mgr.{{ hostvars[item]['ansible_hostname'] }}.keyring
+  changed_when: false
+  when:
+    - cephx
+    - mon_containerized_deployment
+    - groups[mgr_group_name]|length > 0
+    - inventory_hostname == groups[mon_group_name]|last
+    - not mon_containerized_deployment_with_kv
+  with_items: "{{ groups[mgr_group_name] }}"
+>>>>>>> mgr: add new role for ceph-mgr
```
Comment (batrick) [inline roles/ceph-mon/tasks/docker/main.yml]: Patch cruft accidentally merged ^

Comment (leseb) [inline roles/ceph-mon/tasks/docker/main.yml]: Oops, fixed.

**Comment by leseb** [comment]
@batrick thanks, I've also noticed that from the ceph-mgr startup. They don't seem to be a bug issue as the mgr starts. This should be addressed as part of the packages dependencies. @ktdreyer what do you think?

**Comment by ktdreyer** [comment]
@leseb I agree it should be handled as part of the package's dependencies.

@batrick what source(s) did you use to install those? A problem with ceph-mgr packaging right now is that no one has volunteered to maintain the REST stack of dependencies (ie django) for the long term.

**Comment by batrick** [comment]
@ktdreyer This was on CentOS 6 (on Linode). Ceph packages came from shaman via ceph-ansible. I installed python-prettytable etc. via yum.

**Comment by ktdreyer** [comment]
@batrick Was it CentOS 6 or 7?

EPEL 7 has python-django-1.6.11-5.el7, but that [went EOL](https://www.djangoproject.com/weblog/2015/apr/01/release-18-final/) upstream two years ago, so I don't think this is [maintained](https://bugzilla.redhat.com/show_bug.cgi?id=1432365).

It would be a good idea to file a tracker.ceph.com ticket for the django packaging to support ceph-mgr. I think this could happen in the CentOS Storage SIG apart from EPEL. I think Boris Ranto has also done some experiments with using Flask instead.

**Comment by batrick** [comment]
Sorry, CentOS 7.

**Comment by leseb** [comment]
@batrick do you mind filling up a ticket? Thanks!

**Comment by batrick** [comment]
Here's a ticket: http://tracker.ceph.com/issues/19399

**Comment by leseb** [comment]
jenkins test ansible2.2-docker_dmcrypt_journal_collocation

**Comment by leseb** [comment]
jenkins test ansible2.2-docker_dedicated_journal

**Comment by leseb** [comment]
jenkins test ansible2.2-docker_cluster

**Comment by leseb** [comment]
jenkins test this please

**Comment by leseb** [comment]
Jewel failure are expected as there is no ceph-mgr daemon on Jewel.

**Comment by leseb** [comment]
jenkins test kraken-ansible2.2-purge_cluster

**Comment by leseb** [comment]
jenkins test  jewel-ansible2.2-centos7_cluster

**Comment by leseb** [comment]
jenkins test kraken-ansible2.2-centos7_cluster

**Pull request merged**